File: app/src/main/java/com/akshayholla/openweather/data/network/OpenWeatherApi.java
package com.akshayholla.openweather.data.network;

import com.akshayholla.openweather.common.ConstantsKt;
import com.akshayholla.openweather.data.network.model.WeatherDetail;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OpenWeatherApi {
    @GET("/" + ConstantsKt.api25 + "/weather")
    Single<WeatherDetail> getWeatherByCoordinates(
            @Query("lat") double latitude,
            @Query("lon") double longitude,
            @Query("units") String units,
            @Query("appid") String appid
    );

    @GET("/" + ConstantsKt.api25 + "/weather")
    Single<WeatherDetail> getWeatherByCityName(
            @Query("q") String name,
            @Query("units") String units,
            @Query("appid") String appid
    );
}

File: app/src/main/java/com/akshayholla/openweather/data/network/OpenWeatherApi.kt
package com.akshayholla.openweather.data.network

import com.akshayholla.openweather.common.api25
import com.akshayholla.openweather.data.network.model.WeatherDetail
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherApi {
    @GET("/${api25}/weather")
    suspend fun getWeatherByCoordinates(
        @Query("lat") latitude: Double,
        @Query("lon") longitude: Double,
        @Query("units") units: String,
        @Query("appid") appid: String,
    ): Response<WeatherDetail>

    @GET("/${api25}/weather")
    suspend fun getWeatherByCityName(
        @Query("q") name: String,
        @Query("units") units: String,
        @Query("appid") appid: String,
    ): Response<WeatherDetail>
}